app/routes/_gcn.circulars/circulars.lib.ts

```diff
@@ -35,7 +35,10 @@ export interface CircularChangeRequest extends CircularMetadata {
 type SubjectMatcher = [RegExp, (match: RegExpMatchArray) => string]
 
 const subjectMatchers: SubjectMatcher[] = [
-  [/GRB[.\s_-]*(\d{6}[a-z|.]\d*)/i, ([, id]) => `GRB ${id.toUpperCase()}`],
+  [
+    /GRB[.\s_-]*(\d{6,7})([A-Za-z]?|\.\d{1,2})?\S*/i,
+    ([, id, letter]) => `GRB ${id.toUpperCase()}${letter}`,
```

Comment:
```suggestion
    /GRB[.\s_-]*(\d{6}(?:[a-z]?|\.\d{1,2})?)/i,
    ([, id]) => `GRB ${id.toUpperCase()}`,
```

- The `\d{6}` pattern was supposed to match a six-digit date in the format YYMMDD. You replaced it with `\d{6,7}` which will match 7-digit numbers too, but those cannot be valid dates.
- I think that you can safely shorten `[A-Za-z]` to `[a-z]` because you have specified this as a case-insensitive regex (it ends in `/i`).
- Reformat as a single matching group.
- Remove trailing `\S*`. This does nothing.

Reply:
are decimals valid since they aren't 6 digit dates? or should this just get skipped
'Fermi GRB190515.49 Global MASTER-Net observations report'

Reply:
@jracusin?

Reply:
I spoke to Judy on slack. she said we can ignore the MASTER ones for right now